compiler/rustc_target/src/abi/mod.rs

```diff
@@ -234,6 +234,47 @@ pub enum Endian {
     Big,
 }
 
+impl Endian {
+    pub fn name(&self) -> &str {
+        match self {
+            Self::Little => "little",
+            Self::Big => "big",
+        }
+    }
+
+    pub fn is_big(&self) -> bool {
+        Self::Big == *self
+    }
+
+    pub fn is_little(&self) -> bool {
+        Self::Little == *self
+    }
+}
+
+impl fmt::Debug for Endian {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(self.name())
+    }
+}
+
+impl FromStr for Endian {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "little" => Ok(Self::Little),
+            "big" => Ok(Self::Big),
+            _ => Err(format!(r#"unknown endian: "{}""#, s)),
+        }
+    }
+}
+
+impl ToJson for Endian {
+    fn to_json(&self) -> Json {
+        Json::String(self.name().to_owned())
```

Comment:
```suggestion
        self.as_str().to_json()
```

Reply:
It both works the same since I still have to import Json and ToJson
for `impl ToJson for Endian`.